services/dask-sidecar/tests/unit/test_file_utils.py

```diff
@@ -29,118 +34,302 @@ async def mocked_log_publishing_cb(
         yield mocked_callback
 
 
-async def test_copy_file_to_remote(
-    ftpserver: ProcessFTPServer,
+pytest_simcore_core_services_selection = [
+    "postgres"
+]  # TODO: unnecessary but test framework requires it, only minio is useful here
+pytest_simcore_ops_services_selection = ["minio"]
+
+
+@pytest.fixture
+def s3_presigned_link_storage_kwargs(
+    minio_config: dict[str, Any], minio_service: Minio
+) -> dict[str, Any]:
+    return {}
+
+
+@pytest.fixture
+def ftp_remote_file_url(ftpserver: ProcessFTPServer, faker: Faker) -> AnyUrl:
+    return parse_obj_as(
+        AnyUrl, f"{ftpserver.get_login_data(style='url')}/{faker.file_name()}"
+    )
+
+
+@pytest.fixture
+def s3_presigned_link_remote_file_url(
+    minio_config: dict[str, Any],
+    minio_service: Minio,
+    faker: Faker,
+) -> AnyUrl:
+
+    return parse_obj_as(
+        AnyUrl,
+        minio_service.presigned_put_object(
+            minio_config["bucket_name"], faker.file_name()
+        ),
+    )
+
+
+@pytest.fixture
+def s3_remote_file_url(minio_config: dict[str, Any], faker: Faker) -> AnyUrl:
+    return parse_obj_as(
+        AnyUrl, f"s3://{minio_config['bucket_name']}{faker.file_path()}"
+    )
+
+
+@dataclass(frozen=True)
+class StorageParameters:
+    s3_settings: Optional[S3Settings]
+    remote_file_url: AnyUrl
+
+
+@pytest.fixture(params=["ftp", "s3"])
+def remote_parameters(
+    request: FixtureRequest,
+    ftp_remote_file_url: AnyUrl,
+    s3_remote_file_url: AnyUrl,
+    s3_settings: S3Settings,
+) -> StorageParameters:
+    return {
+        "ftp": StorageParameters(s3_settings=None, remote_file_url=ftp_remote_file_url),
+        "s3": StorageParameters(
+            s3_settings=s3_settings, remote_file_url=s3_remote_file_url
+        ),
+    }[
+        request.param  # type: ignore
+    ]
+
+
+async def test_push_file_to_remote(
+    remote_parameters: StorageParameters,
     tmp_path: Path,
     faker: Faker,
     mocked_log_publishing_cb: mock.AsyncMock,
 ):
-    ftp_server_base_url = ftpserver.get_login_data(style="url")
+    # let's create some file with text inside
+    src_path = tmp_path / faker.file_name()
+    TEXT_IN_FILE = faker.text()
+    src_path.write_text(TEXT_IN_FILE)
```

Comment:
@GitHK also uses fake to add lines in fixture files.... but this in not really used for nothing other the "inflate" the file. 
I wonder if it would rather not be more meaningful to parametrize for instance the file sizes and eventually add a small fake "check sentence" at some random places. 

This way we could test how effective is to upload/download files of different sizes and even verify the content in reception.

Reply:
I think `faker.text()` is the one which always gives you some random text (always different from call to call).

From what I understand here, the idea is to test that s3f's uploading works correctly. I think size is not important in this case.

Reply:
the point of that test is indeed to check if it just function.
I would create a separate test for testing performances